libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -340,56 +343,62 @@ void EthStratumClient::connect_handler(const boost::system::error_code& ec, tcp:
 				// Do not trigger a full disconnection but, instead, let the loop
 				// continue with another IP (if any). 
 				// Disconnection is triggered on no more IP available
-
+				m_connected.store(false, std::memory_order_relaxed);
 				m_socket->close();
 				start_connect(++i);
 				return;
 			}
 		}
 
-		// This should be done *after* a valid connection which may fail
-		// on secure connection.
-		m_connected.store(true, std::memory_order_relaxed);
+		// Trigger event handlers and begin counting for the next job
 		if (m_onConnected) { m_onConnected(); }
-
-		// Successfully connected so we start our work timeout timer
 		reset_work_timeout();
 
 		string user;
 		size_t p;
 
+		Json::Value jReq;
```

Comment:
most requests are within json rpc spec (id, method, jsonrpc and params) some also contain worker which is not really within spec. Cant we just use a method containing all 5 elements as arguments. I think this would be safer to not "forget" one element.  id, method and jsonrpc are mandatory.

Reply:
We could with very little effort.
Nevertheless, for example, member "jsonrpc"="2.0" is not applicable to ETHEREUMSTRATUM and there is one case where EthStratumClient acts as server when replying to client.get_version : in that case method member should not be present.